### ISSUE TITLE
Could you export a framework that has .modulemap file?

### ISSUE THREAD

**Issue by JeroldLiu777** [gna-sw/Jabra-Perform-BlueParrott-iOS-SDK#1]
At present, it not contain a .modulemap file which cause a integrate issue from my side. Thanks

**Comment by alanslattery** [comment]
Can you provide some more details on the kind of project you are trying to integrate the SDK with? (i.e. is it swift-based, objective-c etc.) Can you also provide some more information on the error you are getting?

**Comment by JeroldLiu777** [comment]
> 

yes, I am using a swift project, I cannot find the head file. I try to import your module, but it doesn't work without modulemap

**Comment by alanslattery** [comment]
I tried following the instructions here: https://welcm.uk/blog/connecting-an-objective-c-framework-to-your-swift-app
This allowed me to import the framework into a Swift App by just adding a bridging header to the app's project.
In the bridging header, I just added:
`#import <BPHeadset.h>`

Then, I set the filename of the bridging header as the "Objective-C Bridging Header" in the "Swift Compiler - General" settings.
After that I was able to access all the BPHeadset methods, and I could declare my ViewController like this:
```
class ViewController: UIViewController, BPHeadsetListener {
  private var headset: BPHeadset?

  override func viewDidLoad() {
    headset = BPHeadset.sharedInstance()
    headset?.add(self)
    super.viewDidLoad()
  }

```
After that, I could run the app and receive events from the BPHeadset class.
Is this a possible workaround for you, or is something preventing this approach?

**Comment by JeroldLiu777** [comment]
> I tried following the instructions here: https://welcm.uk/blog/connecting-an-objective-c-framework-to-your-swift-app This allowed me to import the framework into a Swift App by just adding a bridging header to the app's project. In the bridging header, I just added: `#import <BPHeadset.h>`
> 
> Then, I set the filename of the bridging header as the "Objective-C Bridging Header" in the "Swift Compiler - General" settings. After that I was able to access all the BPHeadset methods, and I could declare my ViewController like this:
> 
> ```
> class ViewController: UIViewController, BPHeadsetListener {
>   private var headset: BPHeadset?
> 
>   override func viewDidLoad() {
>     headset = BPHeadset.sharedInstance()
>     headset?.add(self)
>     super.viewDidLoad()
>   }
> ```
> 
> After that, I could run the app and receive events from the BPHeadset class. Is this a possible workaround for you, or is something preventing this approach?

Many thanks, yes, indeed, it works with a bridge header file, but in my project, it is not recommended. We use the .modulemap instead. Code reviews may fail. So I seek your help.

**Comment by JeroldLiu777** [comment]
We can't directly add a modulemap file to your SDK, we need your help. This feature is very important to us, please.

**Comment by gna-sw** [comment]
> 

Hello Lu,  I am going to email you directly.

**Comment by JeroldLiu777** [comment]
> 

Thanks!

**Comment by gna-sw** [comment]
> > 
> 
> Thanks!

Hello @lucifer717  did you get the email?  Did you have a better email for us to contact you with?

**Comment by JeroldLiu777** [comment]
> 

Yes, I got your email from jerold717@outlook.com. I have answered your email, please check it.

**Comment by JeroldLiu777** [comment]
> > > 
> > 
> > 
> > Thanks!
> 
> Hello @lucifer717 did you get the email? Did you have a better email for us to contact you with?

And what do you mean by a 'better' email? Official email from my organization？
